tests/integration/test_charm_metrics.py

```diff
@@ -337,3 +368,88 @@ async def test_charm_issues_metrics_for_failed_runner(
         github_repository=forked_github_repository,
         post_job_status=PostJobStatus.REPO_POLICY_CHECK_FAILURE,
     )
+
+
+@pytest.mark.asyncio
+@pytest.mark.abort_on_fail
+async def test_charm_issues_metrics_for_abnormal_termination(
+    model: Model,
+    app: Application,
+    forked_github_repository: Repository,
+    forked_github_branch: Branch,
+):
+    """
+    arrange: A properly integrated charm with a runner registered on the fork repo.
+    act: Dispatch a test workflow and afterwards kill run.sh. After that, reconcile.
+    assert: The RunnerStart, RunnerStop and Reconciliation metric is logged.
+        The Reconciliation metric has the post job status set to Abnormal.
+    """
+    await app.set_config({"path": forked_github_repository.full_name})
+    await ensure_charm_has_runner(app=app, model=model)
+
+    unit = app.units[0]
+
+    workflow = forked_github_repository.get_workflow(
+        id_or_file_name=DISPATCH_CRASH_TEST_WORKFLOW_FILENAME
+    )
+    assert workflow.create_dispatch(forked_github_branch, {"runner": app.name})
+
+    # Wait for the workflow log to be picked up by the runner
+    sleep(60)
```

Comment:
Is there any other way we could use to determine whether it has actually picked up the log? This way if it takes shorter we waste time, if it takes longer we get an unexpected failure

Reply:
One idea might be to inspect the job logs? Or to check whether the runner has started to execute a job?